### DISCUSSION TITLE
Gunshot sound layering

### DISCUSSION THREAD

**Discussion by seaz5150** [OGSR/OGSR-Engine#287] [Ideas]
I would love to see something like [this](https://www.youtube.com/watch?v=4u1OmqEUfOw) made possible in SoC by adding the sound layering included in OpenXRray. Also gunshots use different sound sections for the actor and NPCs.

**Comment by xrSimpAI** [comment]
> adding the sound layering included in OpenXRray

I don't really understand what this is about. Can you show a commit or a place in the OpenXRay code where this is implemented?

**Comment by seaz5150** [comment]
I cant find a commit where this was originally implemented but I found this fix for it: https://github.com/OpenXRay/xray-16/pull/225

**Comment by seaz5150** [comment]
Or I guess it is the original commit, because it says "LAYERED_SND_SHOOT from COC was fully transferred".

**Comment by xrSimpAI** [comment]
Ok, I'll review this.

**Comment by seaz5150** [comment]
Also in the addon different sounds are played at different distances. Not sure if this is implemented in the engine somehow or what but I didnt see any scripts for it. Would have to look into that more.

**Comment by rutrash** [comment]
Sounds in the distance were not only in this addon, It was in standard Anomaly too.

**Comment by seaz5150** [comment]
Damn, I have no idea where to find the distance attenuation in the source files. It might not be included in OpenXRay maybe, but it is in CoC.

**Comment by xrSimpAI** [comment]
Perhaps there is something similar in the SWM repo, as far as I remember, I saw there a lot of commits related to the sounds of weapons (echo, etc.)

**Comment by seaz5150** [comment]
A guy from the OpenXRay discord just sent me [this commit](https://github.com/revolucas/CoC-Xray/commit/c3f83363db619738e8d9f2c999cd8f783daaf719), apparently the different sounds for distances is done through ogg comments, so only implementing the sound layers and seperate sounds for actor is enough.

**Comment by seaz5150** [comment]
Hello, did you have a chance to check out the commits? Is this a feature you would like to add/would it be possible to add it? Cheers.

**Comment by xrSimpAI** [comment]
I'll check it sometime. Now I'm too lazy to do this.
